.github/scripts/Dockerfile

```diff
@@ -0,0 +1,54 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+FROM fedora:34 as builder
+
+# FROM quay.io/centos/centos:stream8 as builder
+# Enable additional package repositories for CentOS
+#  note: PowerTools is called CodeReady Linux Builder in RHEL 8
+# RUN dnf -y install epel-release && dnf config-manager --set-enabled powertools && dnf -y install gcc gcc-c++ make cmake cyrus-sasl-devel cyrus-sasl-plain cyrus-sasl-gssapi cyrus-sasl-md5 openssl-devel libuuid-devel nodejs swig wget patch findutils git valgrind libwebsockets-devel python3-devel libnghttp2-devel && dnf clean all -y
+RUN dnf -y install gcc gcc-c++ make cmake cyrus-sasl-devel cyrus-sasl-plain cyrus-sasl-gssapi cyrus-sasl-md5 openssl-devel libuuid-devel nodejs swig wget patch findutils git valgrind libwebsockets-devel python3-devel libnghttp2-devel && dnf clean all -y
+WORKDIR /build
+COPY . .
+ENV PROTON_VERSION=0.36.0
+ENV PROTON_SOURCE_URL=${PROTON_SOURCE_URL:-http://archive.apache.org/dist/qpid/proton/${PROTON_VERSION}/qpid-proton-${PROTON_VERSION}.tar.gz}
+RUN .github/scripts/compile.sh
+
+
+FROM fedora:34
```

Comment:
@kgiusti @ganeshmurthy Is there any reason why our libwebsockets (in downstream) is built with `LWS_WITH_LIBUV=ON` and `LWS_WITH_HTTP2=OFF`? Thing is, libwebsockets has multiple backend options, libuv is only one of multiple event loops possible. Why was this one chosen?

(I tried asking for libwebsockets be added to ubi8 image, but I got rebuffed, since the library is not part of standard rhel8 either. https://bugzilla.redhat.com/show_bug.cgi?id=2051603)